arch/mips/core/isr.S

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2017 Imagination Technologies Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+# Keep each function in a separate named section
+#define _FUNCTION_SECTIONS_
+
+#include <toolchain.h>
+#include <kernel_structs.h>
+#include <offsets_short.h>
+
+#include <mips/regdef.h>
+#include <mips/asm.h>
+#include <mips/cpu.h>
+#include <mips/hal.h>
+
+GDATA(_sw_isr_table)
+
+GTEXT(_k_neg_eagain)
+
+LEAF(__isr_vec)
+    .set 	push
+    .set 	noat
+AENT(__isr_vec_single)
+    j 	_mips_interrupt
+    .set 	pop
+END(__isr_vec)
+
+#define OS_STACK_ALIGNMENT (8)
+
+#define CTX_ALIGNED_SIZE ((((CTX_SIZE - 1) / OS_STACK_ALIGNMENT) + 1) * \
+    OS_STACK_ALIGNMENT)
+
+LEAF(_mips_interrupt)
+    .set    noat
+    /* Context save */
+    addi    sp, sp, -CTX_ALIGNED_SIZE
+    REG_S   $1, CTX_REG(1)(sp)
+    REG_S   $2, CTX_REG(2)(sp)
+    REG_S   $3, CTX_REG(3)(sp)
+    REG_S   $4, CTX_REG(4)(sp)
+    REG_S   $5, CTX_REG(5)(sp)
+    REG_S   $6, CTX_REG(6)(sp)
+    REG_S   $7, CTX_REG(7)(sp)
+    REG_S   $8, CTX_REG(8)(sp)
+    REG_S   $9, CTX_REG(9)(sp)
+    REG_S   $10, CTX_REG(10)(sp)
+    REG_S   $11, CTX_REG(11)(sp)
+    REG_S   $12, CTX_REG(12)(sp)
+    REG_S   $13, CTX_REG(13)(sp)
+    REG_S   $14, CTX_REG(14)(sp)
+    REG_S   $15, CTX_REG(15)(sp)
+    REG_S   $16, CTX_REG(16)(sp)
+    REG_S   $17, CTX_REG(17)(sp)
+    REG_S   $18, CTX_REG(18)(sp)
+    REG_S   $19, CTX_REG(19)(sp)
+    REG_S   $20, CTX_REG(20)(sp)
+    REG_S   $21, CTX_REG(21)(sp)
+    REG_S   $22, CTX_REG(22)(sp)
+    REG_S   $23, CTX_REG(23)(sp)
+    REG_S   $24, CTX_REG(24)(sp)
+    REG_S   $25, CTX_REG(25)(sp)
+    REG_S   $26, CTX_REG(26)(sp)
+    REG_S   $27, CTX_REG(27)(sp)
+    REG_S   $28, CTX_REG(28)(sp)
+    REG_S   $29, CTX_REG(29)(sp)
+    REG_S   $30, CTX_REG(30)(sp)
+    REG_S   $31, CTX_REG(31)(sp)
+    PTR_S   $0, CTX_LINK(sp)        # Clear the link field
+
+    #if (__mips_isa_rev < 6)
+    mfhi	t0
+    mflo	t1
+    REG_S	t0, CTX_HI0(sp)
+    REG_S	t1, CTX_LO0(sp)
+    #endif
+
+    # cp0
+    PTR_MFC0 t0, C0_EPC
+    REG_S	t0, CTX_EPC(sp)
+    PTR_MFC0 t1, C0_BADVADDR
+    REG_S	t1, CTX_BADVADDR(sp)
+    mfc0	t0, C0_SR
+    sw	    t0, CTX_STATUS(sp)
+    mfc0    t1, C0_CR
+    sw      t1, CTX_CAUSE(sp)
+
+    move    t4, zero
+    move	t5, zero
+    mfc0	t2, C0_CONFIG3
+    ext	    t3, t2, CFG3_BP_SHIFT, 1
```

Comment:
please use consistent spacing between ASM mnemonics and their parameters, it is all over the place in this and other asm files. on other arches we typically use a tab.